arwen/elrondapi/elrondei.go

```diff
@@ -600,8 +614,15 @@ func setAsyncContextCallback(context unsafe.Pointer,
 
 //export asyncCall
 func asyncCall(context unsafe.Pointer, destOffset int32, valueOffset int32, dataOffset int32, length int32) {
-	runtime := arwen.GetRuntimeContext(context)
-	metering := arwen.GetMeteringContext(context)
+	host := arwen.GetVmContext(context)
+	runtime := host.Runtime()
+	metering := host.Metering()
+
+	if !host.AreAsyncCallsAllowed() {
```

Comment:
Removed the flag.

This has nothing to do with the mainnet launch. It was supposed to be `false` only within calls to `init()`, to prevent having the SC in an inconsistent state due to a cross-shard async call. It shouldn't be an issue (`callback()` still gets called), but we definitely need tests for cross-shard async calls during `init()`.